cibuildwheel/options.py

```diff
@@ -183,3 +200,205 @@ def __call__(
             return str(result)
         else:
             return result
+
+
+def compute_options(
+    platform: PlatformName,
+    package_dir: Path,
+    output_dir: Path,
+    config_file: Optional[str],
+    args_archs: Optional[str],
+    prerelease_pythons: bool,
+) -> BuildOptions:
+    """
+    Compute the options from the environment and configuration file.
+    """
+
+    manylinux_identifiers = {
+        f"manylinux-{build_platform}-image" for build_platform in MANYLINUX_ARCHS
+    }
+    musllinux_identifiers = {
+        f"musllinux-{build_platform}-image" for build_platform in MUSLLINUX_ARCHS
+    }
+    disallow = {
+        "linux": {"dependency-versions"},
+        "macos": manylinux_identifiers | musllinux_identifiers,
+        "windows": manylinux_identifiers | musllinux_identifiers,
+    }
+    options = ConfigOptions(package_dir, config_file, platform=platform, disallow=disallow)
+
+    build_config = options("build", env_plat=False, sep=" ") or "*"
+    skip_config = options("skip", env_plat=False, sep=" ")
+    test_skip = options("test-skip", env_plat=False, sep=" ")
+
+    prerelease_pythons = prerelease_pythons or strtobool(
+        os.environ.get("CIBW_PRERELEASE_PYTHONS", "0")
+    )
+
+    deprecated_selectors("CIBW_BUILD", build_config, error=True)
+    deprecated_selectors("CIBW_SKIP", skip_config)
+    deprecated_selectors("CIBW_TEST_SKIP", test_skip)
+
+    package_files = {"setup.py", "setup.cfg", "pyproject.toml"}
+
+    if not any(package_dir.joinpath(name).exists() for name in package_files):
+        names = ", ".join(sorted(package_files, reverse=True))
+        msg = f"cibuildwheel: Could not find any of {{{names}}} at root of package"
+        print(msg, file=sys.stderr)
+        sys.exit(2)
+
+    # This is not supported in tool.cibuildwheel, as it comes from a standard location.
+    # Passing this in as an environment variable will override pyproject.toml, setup.cfg, or setup.py
+    requires_python_str: Optional[str] = os.environ.get(
+        "CIBW_PROJECT_REQUIRES_PYTHON"
+    ) or get_requires_python_str(package_dir)
+    requires_python = None if requires_python_str is None else SpecifierSet(requires_python_str)
+
+    build_selector = BuildSelector(
+        build_config=build_config,
+        skip_config=skip_config,
+        requires_python=requires_python,
+        prerelease_pythons=prerelease_pythons,
+    )
+    test_selector = TestSelector(skip_config=test_skip)
+
+    return _compute_single_options(
+        options, args_archs, build_selector, test_selector, platform, package_dir, output_dir
+    )
+
+
+def _compute_single_options(
+    options: ConfigOptions,
+    args_archs: Optional[str],
+    build_selector: BuildSelector,
+    test_selector: TestSelector,
+    platform: PlatformName,
+    package_dir: Path,
+    output_dir: Path,
+) -> BuildOptions:
+    """
+    Gather options from the command line, environment, and configuration file.
```

Comment:
Not specific to this function now that I refactored (and this happened to have be written by Copilot, FYI).

```suggestion
    Compute BuildOptions for a single run configuration.
```

Something like that. Will make more since once overrides can run this multiple times.

Reply:
> and this happened to have be written by Copilot, FYI 

This was confusing me quite a bit